### PR TITLE
support influxdb with ingress

### PR DESCRIPTION
**Pull request by mllu**
**What this PR does / why we need it**:
Trying to support ingress for InfluxDB, by default it's disabled, but leave option to ppl

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

### CODE REVIEW

File: stable/influxdb/templates/ingress.yaml [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    release: "{{ .Release.Name }}"
+    heritage: "{{ .Release.Service }}"
+  annotations:
+{{ toYaml .Values.ingress.annotations | indent 4 }}
+spec:
+{{- if .Values.ingress.tls }}
+  tls:
+    - hosts:
+      - {{ .Values.ingress.hostname | quote }}
+      secretName: {{ .Values.ingress.secretName | default (printf "%s-tls" (include "influxdb.fullname" .)) }}
```
Comment (unguiculus) [inline stable/influxdb/templates/ingress.yaml]: - This doesn't seem to make sense. Why would you want to use the secret for the DB credentials here?
- `secretName` should be `existingSecret`

**Review by unguiculus — CHANGES_REQUESTED** [review]

File: stable/influxdb/templates/ingress.yaml (hunk 1 above)
Comment (mllu) [reply stable/influxdb/templates/ingress.yaml]: I was trying to reference some existing `ingress` setting like `chronograf` one 
https://github.com/kubernetes/charts/blob/fd93bff82fc18c33dc21e46848af9556c298af1d/stable/chronograf/templates/ingress.yaml#L18

This `secret` is for `ingress` TLS that shows in values.yaml below

Comment (unguiculus) [reply stable/influxdb/templates/ingress.yaml]: Forget about my comment regarding `secretName`. That's fine. But just blindly copying stuff doesn't make it right. It seems Chronograph is broken in that regard. It's fine if you just do `secretName: {{ .Values.ingress.secretName }}`.

**Review by unguiculus — CHANGES_REQUESTED** [review]

File: stable/influxdb/templates/ingress.yaml (hunk 1 above)
Comment (mllu) [reply stable/influxdb/templates/ingress.yaml]: thanks for the feedback, removed the default value

**Review by unguiculus — APPROVED** [review]

**Pull request merged**